manim/animation/creation.py

```diff
@@ -476,7 +476,7 @@ def interpolate_mobject(self, alpha: float) -> None:
         for shape in self.shapes:
             shape.restore()
             shape.save_state()
-            opacity = shape.get_fill_opacity()
+            opacity = shape.get_fill_opacity() or 1
```

Comment:
I'm not sure this is the correct solution here as it modifies the fill opacity of a given mobject -- this actually also happens in the test that you added; at the end of `SpiralIn`, the mobjects should be added to the scene with the opacity values they have been initialized with.

My suggestion for a slightly more complicated fix would be to split the current `opacity` into the typical `stroke_opacity` and `fill_opacity`, and then update both of them according to the rule given in `new_opacity`.

That way, no unexpected opacity changes happen.

Reply:
Howdy @behackl ! Thank you for the feedback. I like your idea much more. I wasn't aware the mobject attributes could be set separately like that. That __getattr__ function is nifty! 
I changed the fix so it lines up with your idea as I understand it, and I updated my test data to reflect that change.

One issue is it looks like my change did throw off the test_banner test. The frame comparison shows that the change causes the starting square to show on the top of the video. Do you think my fix is bad, is there something new to track down, or do we just need to regenerate the test data for test_banner?